src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -44,7 +44,63 @@ public static class IServiceCollectionExtensions
             Guard.NotNull(implementationFactory, nameof(implementationFactory), $"Requires a factory function to create the '{nameof(IAppName)}' implementation");
 
             return services.AddSingleton(implementationFactory)
-                           .AddSingleton<ITelemetryInitializer, ApplicationTelemetryInitializer>();
+                           .AddSingleton<ITelemetryInitializer, ApplicationNameTelemetryInitializer>();
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IAppVersion"/> implementation to the application which uses the current application's assembly version as application version (see: <see cref="AssemblyAppVersion"/>).
+        /// </summary>
+        /// <typeparam name="TConsumerType">Some random consumer type that will be used to retrieve the assembly where the project is running.</typeparam>
+        /// <param name="services">The collection of registered services to add the <see cref="IAppVersion"/> implementation to.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
+        public static IServiceCollection AddAssemblyAppVersion<TConsumerType>(this IServiceCollection services)
```

Comment:
We should also update the documentation and mention that it is possible to enrich with application version, no ?

Reply:
We already do this, no? https://observability.arcus-azure.net/Features/telemetry-enrichment#version-enricher